client/web/compose/src/components/Public/Record/Modal.vue

```diff
@@ -121,12 +125,30 @@ export default {
     ...mapActions({
       setRecordPaginationUsable: 'ui/setRecordPaginationUsable',
       clearRecordIDs: 'ui/clearRecordIDs',
+      pushModalPreviousPage: 'ui/pushModalPreviousPage',
+      clearModalPreviousPage: 'ui/clearModalPreviousPage',
+      popModalPreviousPage: 'ui/popModalPreviousPage',
     }),
 
-    loadRecord ({ recordID, recordPageID, values }) {
+    loadRecord ({ recordID, recordPageID, values, pushModalPreviousPage = true }) {
       this.values = values
       this.loadModal({ recordID, recordPageID })
 
+      // Push the previous modal view page to the modal route history stack on the store so we can go back to it
+      const { prevRecordID, prevRecordPageID } = this.prevRouteQuery
+
+      if (prevRecordID && prevRecordPageID && pushModalPreviousPage) {
+        this.pushModalPreviousPage({ recordID: prevRecordID, recordPageID: prevRecordPageID })
+
+        this.prevRouteQuery = { prevRecordID: recordID, prevRecordPageID: recordPageID }
+      } else {
+        this.popModalPreviousPage()
```

Comment:
As mentioned below, this should be popped in the view/Record.
All the modal does is process the event. Similar to how handle-record-redirect works